### PR TITLE
CORE-13335: Fix examples that assume invalid data cannot be ALIVE

### PR DESCRIPTION
**Pull request by samuelraeburn**
### Summary
Update the keyed_data example that assumes that !valid_data means an instance cannot be ALIVE.

### Details and comments
The introduction of Instance State Consistency means that a sample without valid_data can be in any instance state.
The following code is no longer correct:
```
if (!valid_data)
     if (instance_state == NOT_ALIVE_DISPOSED) {
         // handle disposed
     } else {
         // handle NOT_ALIVE_NO_WRITERS
     }
```

### Checks

-   [x] I have updated the documentation accordingly.
-   [x] I have read the [CONTRIBUTING](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/CONTRIBUTING.md) document.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -168,7 +168,7 @@ pipeline {
                         sh 'tar zxvf connextdds-staging-${CONNEXTDDS_ARCH}.tgz unlicensed/'
 
                         sh '''
-                            cp ${RTI_INSTALLATION_PATH}/rti_connext_dds-*/lib/${CONNEXTDDS_ARCH}/openssl-1.*/* \
+                            cp ${RTI_INSTALLATION_PATH}/rti_connext_dds-*/lib/${CONNEXTDDS_ARCH}/openssl-3.*/* \
```
Comment (samuelraeburn) [inline Jenkinsfile:171]: @ManuelJNunez asked me to make this change. It should fix the CI.

**Comment by samuelraeburn** [comment]
Failure is not in my changes. The coherent_presentation example is broken:
```
[ 10%] Building CXX object connext_dds/coherent_presentation/c++11/CMakeFiles/coherent_presentation_subscriber_cpp2.dir/coherent_subscriber.cxx.o
/rti/jenkins/workspace/ci_rticonnextdds-examples_PR-645@2/examples/connext_dds/coherent_presentation/c++11/coherent_subscriber.cxx: In function ‘void run_subscriber_application(unsigned int, unsigned int)’:
/rti/jenkins/workspace/ci_rticonnextdds-examples_PR-645@2/examples/connext_dds/coherent_presentation/c++11/coherent_subscriber.cxx:97:26: error: ‘shutdown_requested’ is not a member of ‘application’
   97 |     while (!application::shutdown_requested && samples_read < sample_count) {
      |                          ^~~~~~~~~~~~~~~~~~
/rti/jenkins/workspace/ci_rticonnextdds-examples_PR-645@2/examples/connext_dds/coherent_presentation/c++11/coherent_subscriber.cxx: In function ‘int main(int, char**)’:
/rti/jenkins/workspace/ci_rticonnextdds-examples_PR-645@2/examples/connext_dds/coherent_presentation/c++11/coherent_subscriber.cxx:116:5: error: ‘setup_signal_handlers’ was not declared in this scope
  116 |     setup_signal_handlers();
      |     ^~~~~~~~~~~~~~~~~~~~~
make[2]: *** [connext_dds/coherent_presentation/c++11/CMakeFiles/coherent_presentation_subscriber_cpp2.dir/build.make:63: connext_dds/coherent_presentation/c++11/CMakeFiles/coherent_presentation_subscriber_cpp2.dir/coherent_subscriber.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:5540: connext_dds/coherent_presentation/c++11/CMakeFiles/coherent_presentation_subscriber_cpp2.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

**Review by alexcamposruiz — APPROVED** [review]

**Comment by alexcamposruiz** [comment]
The destination branch should be master, not develop.

**Comment by alexcamposruiz** [comment]
> Failure is not in my changes. The coherent_presentation example is broken:
> 
> ```
> [ 10%] Building CXX object connext_dds/coherent_presentation/c++11/CMakeFiles/coherent_presentation_subscriber_cpp2.dir/coherent_subscriber.cxx.o
> /rti/jenkins/workspace/ci_rticonnextdds-examples_PR-645@2/examples/connext_dds/coherent_presentation/c++11/coherent_subscriber.cxx: In function ‘void run_subscriber_application(unsigned int, unsigned int)’:
> /rti/jenkins/workspace/ci_rticonnextdds-examples_PR-645@2/examples/connext_dds/coherent_presentation/c++11/coherent_subscriber.cxx:97:26: error: ‘shutdown_requested’ is not a member of ‘application’
>    97 |     while (!application::shutdown_requested && samples_read < sample_count) {
>       |                          ^~~~~~~~~~~~~~~~~~
> /rti/jenkins/workspace/ci_rticonnextdds-examples_PR-645@2/examples/connext_dds/coherent_presentation/c++11/coherent_subscriber.cxx: In function ‘int main(int, char**)’:
> /rti/jenkins/workspace/ci_rticonnextdds-examples_PR-645@2/examples/connext_dds/coherent_presentation/c++11/coherent_subscriber.cxx:116:5: error: ‘setup_signal_handlers’ was not declared in this scope
>   116 |     setup_signal_handlers();
>       |     ^~~~~~~~~~~~~~~~~~~~~
> make[2]: *** [connext_dds/coherent_presentation/c++11/CMakeFiles/coherent_presentation_subscriber_cpp2.dir/build.make:63: connext_dds/coherent_presentation/c++11/CMakeFiles/coherent_presentation_subscriber_cpp2.dir/coherent_subscriber.cxx.o] Error 1
> make[1]: *** [CMakeFiles/Makefile2:5540: connext_dds/coherent_presentation/c++11/CMakeFiles/coherent_presentation_subscriber_cpp2.dir/all] Error 2
> make: *** [Makefile:84: all] Error 2
> ```

this is fixed in master. I've merged master into develop so it's now also fixed in develop.

**Review by alexcamposruiz — CHANGES_REQUESTED** [review]
Sync your repo and update destination branch.

**Review by alexcamposruiz — CHANGES_REQUESTED** [review]
the commit brings unrelated changes to master (dynamic_data_skip_serialization)

**Comment by ManuelJNunez** [comment]
Retarget to a branch other than master please. For example release/7.3.0

**Comment by samuelraeburn** [comment]
> Retarget to a branch other than master please. For example release/7.3.0

Can we create some guidelines as to what the target branches for this repo should be? I was following this task: https://github.com/rticommunity/rticonnextdds-examples/pull/645#issuecomment-1912352411 and it has held up the PR for weeks

**Comment by samuelraeburn** [comment]
Creating a new PR since this one is a mess: https://github.com/rticommunity/rticonnextdds-examples/pull/668

**Pull request closed without merging**